target/linux/generic/files/drivers/mtd/mtdsplit/mtdsplit_uimage.c

```diff
@@ -376,6 +387,65 @@ static struct mtd_part_parser uimage_edimax_parser = {
 	.type = MTD_PARSER_TYPE_FIRMWARE,
 };
 
+
+/**************************************************
+ * Fon(Foxconn)
+ **************************************************/
+
+#define FONFXC_PAD_LEN		32
+
+static ssize_t uimage_find_fonfxc(u_char *buf, size_t len, int *extralen)
+{
+	struct uimage_header *header = (struct uimage_header *)buf;
+	uint32_t crc_orig;
+	uint32_t crc_calc;
+	ssize_t	ret = -EINVAL;
+
+	if (uimage_verify_default(buf, len, extralen) < 0)
+		return -EINVAL;
+
+	crc_orig = be32_to_cpu(header->ih_hcrc);
+
+	header->ih_hcrc = 0;
+	crc_calc = uimage_crc32(0, buf,	sizeof(*header) + FONFXC_PAD_LEN);
```

Comment:
I don't think you need to calculate and verify the crc at this point anymore. Like the other generic uImage parser and the Edimax you can just call uimage_verify_default() and leave it at that.

Reason is that the u-boot must have checked the correctness already otherwise it wouldn't have booted the kernel.

Reply:
If omitting header CRC check, we can not distinguish generic uimage header and fonfxc uimage header.
If actual length of ih_name string is less than 32 bytes, their only differences are the existence of zero padding (offset: +64-+95) and header CRC value.
The other header fields are all same means and same values.

Reply:
> If omitting header CRC check, we can not distinguish generic uimage header and fonfxc uimage header.

Then something (else) is wrong :confused: . The special compatible "fonfxc,uimage" is there to tell the mtdsplit which method to use. https://lists.openwrt.org/pipermail/openwrt-devel/2018-August/013558.html

so, looks like more work is required here. Question is: how to continue?

Reply:
May we assume that CONFIG_OF is enabled, in short, device tree bindings are activated ?
If so, only the parser for "fonfxc,uimage" can be directly called. In that case, CRC check is unnecessary, as you mentioned.

Reply:
Removed CRC check.